packages/core/src/extension-point.ts

```diff
@@ -0,0 +1,116 @@
+// Copyright IBM Corp. 2019. All Rights Reserved.
+// Node module: @loopback/core
+// This file is licensed under the MIT License.
+// License text available at https://opensource.org/licenses/MIT
+
+import {
+  bind,
+  BindingFromClassOptions,
+  BindingSpec,
+  BindingTemplate,
+  Constructor,
+  Context,
+  createBindingFromClass,
+  createViewGetter,
+  filterByTag,
+  inject,
+} from '@loopback/context';
+import {CoreTags} from './keys';
+
+/**
+ * Decorate a class as a named extension point. If the decoration is not
+ * present, the name of the class will be used. For example:
+ *
+ * ```ts
+ * import {extensionPoint} from '@loopback/core';
+ *
+ * @extensionPoint(GREETER_EXTENSION_POINT_NAME)
+ * export class GreetingService {
+ *   // ...
+ * }
+ * ```
+ *
+ * @param name Name of the extension point
+ */
+export function extensionPoint(name: string, ...specs: BindingSpec[]) {
+  const tags = name ? {tags: {name}} : {};
+  return bind(tags, ...specs);
+}
+
+/**
+ * Shortcut to inject extensions for the given extension point. For example:
+ *
+ * ```ts
+ * import {Getter} from '@loopback/context';
+ * import {extensionPoint, extensions} from '@loopback/core';
+ *
+ * @extensionPoint(GREETER_EXTENSION_POINT_NAME)
+ * export class GreetingService {
+ *  constructor(
+ *    @extensions() // Inject extensions for the extension point
+ *    private getGreeters: Getter<Greeter[]>,
+ *    // ...
+ * ) {
+ *   // ...
+ * }
+ * ```
+ *
+ * @param extensionPointName Name of the extension point. If not supplied, we use
+ * the name from `@extensionPoint` or the class name of the extension point
+ * class.
+ */
+export function extensions(extensionPointName?: string) {
+  return inject('', {decorator: '@extensions'}, (ctx, injection, session) => {
+    // Find the key of the target binding
+    if (!session.currentBinding) return undefined;
+
+    if (!extensionPointName) {
+      extensionPointName = session.currentBinding.tagMap.name;
+      if (!extensionPointName) {
+        let target: Function;
+        if (typeof injection.target === 'function') {
+          // Constructor injection
+          target = injection.target;
+        } else {
+          // Injection on the prototype
+          target = injection.target.constructor;
+        }
+        extensionPointName = target.name;
+      }
+    }
+
+    const bindingFilter = filterByTag({
+      [CoreTags.EXTENSION_POINT]: extensionPointName,
+    });
+    return createViewGetter(ctx, bindingFilter, session);
+  });
+}
+
+/**
+ * A factory function to create binding template for extensions of the given
+ * extension point
+ * @param extensionPointName Name of the extension point
+ */
+export function extensionFor(extensionPointName: string): BindingTemplate {
+  return binding => binding.tag({extensionPoint: extensionPointName});
```

Comment:
Uh oh, so an extension is decorated with a tag called `extensionPoint` which means that the decorated class is not an extension point itself, but it's an extension for the specified `extensionPoint`. I find it difficult to get my head around this naming convention. Could you please try to find a better one? Perhaps something along the lines of `targetExtensionPoint` or `extensionFor`?

Reply:
`extensionFor` is now used.